src/apps/shared/commands/encryption.py

```diff
@@ -0,0 +1,146 @@
+    "user_applet_accesses": ["nickname"],
+    "users_workspaces": [
+        "workspace_name",
+        "database_uri",
+        "storage_type",
+        "storage_access_key",
+        "storage_secret_key",
+        "storage_region",
+        "storage_url",
+        "storage_bucket",
+    ],
+    "transfer_ownership": ["email"],
+    "alerts": ["alert_message"],
+    "answer_notes": ["note"],
+    "invitations": ["email", "first_name", "last_name", "nickname"],
+}
+
+
+def print_data_table() -> None:
+    table = Table(
+        *("Table name", "List of encrypted columns"),
+        title="Tables with encrypted columns",
+        title_style=Style(bold=True),
+    )
+    for k, v in TABLE_NAME_COLUMN_NAME_MAP.items():
+        table.add_row(f"[bold]{k}[/bold]", "\n".join(v), end_section=True)
+
+    print(table)
+
+
+@app.command(short_help="Show tables with columns which have ecnrypted data")
+@coro
+async def show() -> None:
+    print_data_table()
+
+
+@app.command(short_help="Reencrypt data")
+@coro
+async def reencrypt(
+    decrypt_secret_key: Optional[str] = typer.Option(
+        None,
+        "--decrypt-secret-key",
+        "-d",
+        help="Hex of secret key to decrypt data, "
+        "if not key specified, default settings key will be used.",
+    ),
+    encrypt_secret_key: Optional[str] = typer.Option(
+        None,
+        "--encrypt-secret-key",
+        "-e",
+        help="Hex of secret key to encrypt data, "
+        "if not key specified, default settings key will be used.",
+    ),
+    tables: Optional[list[str]] = typer.Argument(
+        None,
+        help="Tables which data need to reencrypt, "
+        "if no table names are provided data in ALL allowed tables "
+        "will be reencrypted.",
+    ),
+) -> None:
+    session_maker = session_manager.get_session()
+    decrypt_key = (
+        bytes.fromhex(decrypt_secret_key) if decrypt_secret_key else get_key()
+    )
+    decryptor = StringEncryptedType(Unicode, decrypt_key)
+    encrypt_key = (
+        bytes.fromhex(encrypt_secret_key) if encrypt_secret_key else get_key()
+    )
+    encryptor = StringEncryptedType(Unicode, encrypt_key)
+    tables = tables if tables else list(TABLE_NAME_COLUMN_NAME_MAP.keys())
+    try:
+        async with session_maker() as session:
+            async with atomic(session):
+                for table_name in tables:
+                    columns = TABLE_NAME_COLUMN_NAME_MAP.get(table_name, [])
+                    if not columns:
+                        print(
+                            "[red]"
+                            f"[bold]{table_name}[/bold] table does not have "
+                            "encrypted columns. Skipped[red]"
+                        )
+                        continue
+                    cols = ", ".join(columns)
+                    result = await session.execute(
+                        f"select id, {cols} from {table_name}"
+                    )
+                    for row in result:
```

Comment:
Please add something to control memory, e.g. pagination, initial query conditions, etc